test/mock_compatibility.js

```diff
@@ -0,0 +1,87 @@
+// For the convenience of unit testing, add these to the global namespace
+var _      = require('underscore');
+var expect = require('chai').expect;
+
+// Unit testing is done against the debug version of the library as it has
+// additional conformance checks that are optimized out of the production
+// library.  Again, globals are used purely for convenience.
+var opentracing = require('../debug.js');
+
+module.exports = function mockCompatibilityChecks() {
+    describe('Mock Opentracing Implementation', function() {
+        describe('Constants', function() {
+            var constStrings = [
+                'FORMAT_TEXT_MAP',
+                'FORMAT_BINARY',
+                'FORMAT_HTTP_HEADERS',
+                'REFERENCE_CHILD_OF',
+                'REFERENCE_FOLLOWS_FROM',
+            ];
+            _.each(constStrings, function(name) {
+                it(name + ' should be a constant string', function() {
+                    expect(opentracing[name]).to.be.a('string');
+                });
+            });
+        });
+
+        describe('Standalone functions', function() {
+            var funcs = [
+                'childOf',
+                'followsFrom',
+                'initGlobalTracer',
+                'globalTracer',
+
+            ];
+            _.each(funcs, function(name) {
+                it(name + ' should be a function', function() {
+                    expect(opentracing[name]).to.be.a('function');
+                });
+            });
+        });
+
+        describe('Tracer#inject', function() {
+            it('should enforce the required carrier types', function() {
+                var tracer = new opentracing.Tracer();
+                var spanContext = tracer.startSpan('test_operation').context();
+                expect(function() { tracer.inject(spanContext, opentracing.FORMAT_TEXT_MAP, ''); }).to.throw(Error);
+                expect(function() { tracer.inject(spanContext, opentracing.FORMAT_TEXT_MAP, 5); }).to.throw(Error);
+
+                expect(function() { tracer.extract(opentracing.FORMAT_BINARY, { buffer : '' }); }).to.throw(Error);
+                expect(function() { tracer.extract(opentracing.FORMAT_BINARY, { buffer : 5 }); }).to.throw(Error);
+            });
+        });
+
+        describe('Span#finish', function() {
+            it('should return undefined', function() {
+                var tracer = new opentracing.Tracer();
```

Comment:
even if there's no explicit use case, I think it'd be a nicer design if there was still a factory method to create the tracer that people could theoretically override.